### PR TITLE
feat: remove d2l-navigation-button-notification-icon

### PR DESCRIPTION
**Pull request by dlockhart**
DO NOT MERGE YET, just preparing this to merge once the usages [in BSI](https://github.com/Brightspace/brightspace-integration/pull/7046) and [the LMS](https://github.com/Brightspace/lms/pull/26157) have been removed.

Removes the no longer used `<d2l-navigation-button-notification-icon>`, which has been replaced by `<d2l-navigation-dropdown-button-icon>`.

There are 2 usages outside the LMS:
- Inside some testing scripts that Tesla has -- I've notified Marius of it and he's ready to update it
- Inside some functions that Learning Solutions has -- I've notified Tony of that as well

### CODE REVIEW

File: d2l-navigation-button-notification-icon.js [hunk 1]
```diff
@@ -1,65 +0,0 @@
-import '@brightspace-ui/core/components/colors/colors.js';
-import './d2l-navigation-button.js';
-import '@brightspace-ui/core/components/icons/icon.js';
-import '@brightspace-ui/button/d2l-button-behavior.js';
-import '@brightspace-ui/polymer-behaviors/d2l-focusable-behavior.js';
```
Comment (svanherk) [inline d2l-navigation-button-notification-icon.js:5]: We're getting there... almost!

File: test/button.visual-diff.js [hunk 2]
```diff
@@ -32,8 +32,6 @@ describe('d2l-navigation-button', () => {
 		{ category: 'icon-text-flip', tests: ['normal', 'hover', 'focus'] },
 		{ category: 'icon-text-hidden', tooltip: true, rectSelector: 'icon-text-hidden-container', tests: ['normal', 'hover', 'focus'] },
 		{ category: 'icon-text-hidden-disabled', tests: ['normal', 'hover', 'focus'] },
-		{ category: 'notification-icon-off', tests: ['normal', 'hover', 'focus'] },
-		{ category: 'notification-icon-on', tests: ['normal', 'hover', 'focus'] },
```
Comment (svanherk) [inline test/button.visual-diff.js:36]: Shouldn't there be deleted visual-diffs here?

File: d2l-navigation-button-notification-icon.js (hunk 1 above)
Comment (dlockhart) [reply d2l-navigation-button-notification-icon.js:5]: Yeah just `<d2l-navigation-button>` and `<d2l-navigation-link>` which are both on my hit-list.

File: test/button.visual-diff.js (hunk 2 above)
Comment (dlockhart) [reply test/button.visual-diff.js:36]: Yep -- looks like visual-diff doesn't create a PR unless there's a failure and orphaned goldens don't cause a failure. I've manually deleted the files.

**Review by svanherk — APPROVED** [review]

**Pull request merged**